nexus_constructor/qentity_utils.py

```diff
@@ -0,0 +1,41 @@
+from typing import List
+
+from PySide2.Qt3DCore import Qt3DCore
+from PySide2.Qt3DRender import Qt3DRender
+from PySide2.QtGui import QColor
+
+
+def set_material_properties(
+    material: Qt3DRender.QMaterial,
+    ambient: QColor,
+    diffuse: QColor,
+    alpha: float = None,
+    remove_shininess: bool = False,
+):
+    """
+    Set the ambient, diffuse, and alpha properties of a material.
+    :param material: The material to be modified.
+    :param ambient: The desired ambient colour of the material.
+    :param diffuse: The desired diffuse colour of the material.
+    :param alpha: The desired alpha value of the material. Optional argument as not all material-types have this
+                  property.
+    :param remove_shininess: Boolean indicating whether or not to remove shininess. This is used for the gnomon.
+    """
+    material.setAmbient(ambient)
+    material.setDiffuse(diffuse)
+
+    if alpha is not None:
+        material.setAlpha(alpha)
+
+    if remove_shininess:
+        material.setShininess(0)
+
+
+def add_qcomponents_to_entity(
```

Comment:
As above you could potentially do similar here where you just pass the root entity and this deals with creating the new entity and then returns it.